src/hardware/mixer.cpp

```diff
@@ -1266,14 +1286,15 @@ void MixerChannel::ConvertSamples(const Type *data, const uint16_t frames,
 		out.emplace_back(out_frame[0]);
 		out.emplace_back(out_frame[1]);
 
-		if (do_zoh_upsampler) {
+		if (resample_method == ResampleMethod::ZeroOrderHoldAndResample) {
```

Comment:
This isn't protected `do_resample`; what if it's false? 

(When I read this, it feels like the `do_resample` is being trumped or ignored)

Reply:
Yeah, this might be a bug, adding the check.